inst/include/Rcpp/exceptions_impl.h

```diff
@@ -0,0 +1,69 @@
+namespace Rcpp {
+#if defined(__GNUC__) || defined(__clang__)
+    #if defined(_WIN32) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__CYGWIN__) || defined(__sun) || defined(_AIX) || defined(__MUSL__) || defined(__HAIKU__) || defined(__ANDROID__)
+        // do nothing
+    #else
+    #include <execinfo.h>
+
+    // Extract mangled name e.g. ./test(baz+0x14)[0x400962]
+    static std::string demangler_one(const char* input) {
+        static std::string buffer;
+        buffer = input;
+        size_t last_open = buffer.find_last_of('(');
+        size_t last_close = buffer.find_last_of(')');
+        if (last_open == std::string::npos ||
+            last_close == std::string::npos) {
+            return input;     // #nocov
+        }
+        std::string function_name = buffer.substr(last_open + 1, last_close - last_open - 1);
+        // Strip the +0x14 (if it exists, which it does not in earlier versions of gcc)
+        size_t function_plus = function_name.find_last_of('+');
+        if (function_plus != std::string::npos) {
+            function_name.resize(function_plus);
+        }
+        buffer.replace(last_open + 1, function_name.size(), demangle(function_name));
+        return buffer;
+    }
+    #endif
+#endif
+
+    // thread-safe; invoked prior to throwing the exception
+    inline void exception::record_stack_trace()
```

Comment:
Should this also accept the filename + line where the exception occurred? (I realize this seems to be otherwise unused but we should avoid API changes if possible)

Reply:
Ah, via the `GET_STACKTRACE` macro. I didn't consider that. We'll see what happens with revdep.

Reply:
Hm, given that you've recommended that `stack_trace` can return R_NilValue, I think there is no need to store filename + line.